browser/ui/webui/brave_new_tab/custom_image_chooser.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_new_tab/custom_image_chooser.h"
+
+#include <utility>
+
+#include "brave/components/l10n/common/localization_util.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/chrome_select_file_policy.h"
+#include "chrome/grit/generated_resources.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/shell_dialogs/selected_file_info.h"
+
+namespace brave_new_tab {
+
+CustomImageChooser::CustomImageChooser(content::WebContents* web_contents)
+    : web_contents_(web_contents),
+      profile_(Profile::FromBrowserContext(web_contents->GetBrowserContext())) {
+}
+
+CustomImageChooser::~CustomImageChooser() = default;
+
+void CustomImageChooser::ShowDialog(ShowDialogCallback callback) {
+  if (callback_) {
+    std::move(callback_).Run({});
+  }
+
+  callback_ = std::move(callback);
+
+  if (dialog_) {
+    return;
+  }
+
+  dialog_ = ui::SelectFileDialog::Create(
+      this, std::make_unique<ChromeSelectFilePolicy>(web_contents_));
+
+  ui::SelectFileDialog::FileTypeInfo file_types;
+  file_types.allowed_paths = ui::SelectFileDialog::FileTypeInfo::NATIVE_PATH;
+  file_types.extensions.push_back(
+      {{FILE_PATH_LITERAL("jpg"), FILE_PATH_LITERAL("jpeg"),
```

Comment:
maybe webp too?

Reply:
Also the idea of people setting their background to an animated gif terrifies me :laughing:

Reply:
Unless there's a requirement change, I guess we should stick with the existing list of file types. 🤷